run-tests.sh

```diff
@@ -61,31 +50,30 @@ fetch() {
 fetch "${kernel}"
 
 if fetch "${selftests}"; then
-  mkdir "${input}/bpf"
-  tar --strip-components=4 -xjf "${tmp_dir}/${selftests}" -C "${input}/bpf"
+  mkdir "${scratch}/bpf"
+  tar --strip-components=4 -xjf "${tmp_dir}/${selftests}" -C "${scratch}/bpf"
 else
   echo "No selftests found, disabling"
 fi
 
+# Pre-build test binary on host, output into scratch directory shared with VM.
+CGO_ENABLED=0 go test -cover -c -o "${scratch}/ebpf.test"
```

Comment:
Unfortunately this only runs the tests for the main library, not the nested packages like `asm`, etc. I think Go doesn't allow compiling multiple package into one binary unfortunately.

Options:
* Make the Go build cache work better. I tried doing this before without much success :(
* Compile all sub packages, invoke them manually from within the VM. Probably a PITA.